### PR TITLE
Docker Image Improvements & Http Routes Entensibility

### PR DESCRIPTION
**Pull request by vjkoskela**
Improvemnets to docker image: i) jvm arguments; ii) logback configuration, iii) updated base image (112 is not yet available). Also includes optional http routes extension.

### CODE REVIEW

File: src/main/docker/Dockerfile [hunk 1]
```diff
@@ -12,22 +12,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM java:8u92-jre-alpine
+FROM java:8u111-jre-alpine
```
Comment (BrandonArp) [inline src/main/docker/Dockerfile]: The "java" base image is deprecated.  Please use the opkenjdk base image instead.

File: src/main/java/com/arpnetworking/http/Routes.java [hunk 2]
```diff
@@ -80,16 +80,19 @@
...
      * @param statusPath The path for the status.
+     * @param supplementalRoutes Optional supplemental routes.
      */
     public Routes(
             final ActorSystem actorSystem,
             final MetricsFactory metricsFactory,
             final String healthCheckPath,
-            final String statusPath) {
+            final String statusPath,
+            final Optional<SupplementalRoutes> supplementalRoutes) {
```
Comment (BrandonArp) [inline src/main/java/com/arpnetworking/http/Routes.java]: How about making this a list?  Having a single SupplementalRoute object doesn't allow for easy composition.

File: src/main/docker/Dockerfile (hunk 1 above)
Comment (vjkoskela) [inline src/main/docker/Dockerfile]: Dang. Will do.

File: src/main/java/com/arpnetworking/http/Routes.java (hunk 2 above)
Comment (vjkoskela) [inline src/main/java/com/arpnetworking/http/Routes.java]: I'm not sure I see the point in providing it at this level. This provides an entry point to hook into the http router, if the customizer wants to federate the routes, they can do so with whatever design pattern (e.g. configuration, injection, scannotation, etc.) that makes sense in their customization. 

Providing multiple supplements could be useful if we or a second-party were to add a supplement and then a third-party wanted to add another. I think that might be overcomplicating this a bit.

**Comment by vjkoskela** [comment]
Applied changes as discussed. Had to lower the coverage as the code I added is minor and the affected classes already have no tests. Other areas added recently have pushed us down to the limit; and those areas should receive some testing.

**Pull request merged**